src/utils/IdealGraphVisualizer/HierarchicalLayout/src/main/java/com/sun/hotspot/igv/hierarchicallayout/HierarchicalStableLayoutManager.java

```diff
@@ -0,0 +1,1471 @@
+        }
+
+        final int offset = X_OFFSET + DUMMY_WIDTH;
+
+        int curY = 0;
+        int curWidth = node.width + reversedDown.size() * offset;
+        for (int pos : reversedDown) {
+            ArrayList<LayoutEdge> reversedSuccs = new ArrayList<>();
+            for (LayoutEdge e : node.succs) {
+                if (e.relativeFrom == pos && reversedLinks.contains(e.link)) {
+                    reversedSuccs.add(e);
+                    e.relativeFrom = curWidth;
+                }
+            }
+
+            ArrayList<Point> startPoints = new ArrayList<>();
+            startPoints.add(new Point(curWidth, curY));
+            startPoints.add(new Point(pos, curY));
+            startPoints.add(new Point(pos, reversedDown.size() * offset));
+            for (LayoutEdge e : reversedSuccs) {
+                reversedLinkStartPoints.put(e.link, startPoints);
+            }
+
+            node.inOffsets.put(pos, -curY);
+            curY += offset;
+            node.height += offset;
+            node.yOffset += offset;
+            curWidth -= offset;
+        }
+
+        int widthFactor = reversedDown.size();
+        node.width += widthFactor * offset;
+
+        int curX = 0;
+        int minX = 0;
+        if (hasReversedDown) {
+            minX = -offset * reversedUp.size();
+        }
+
+        int oldNodeHeight = node.height;
+        for (int pos : reversedUp) {
+            ArrayList<LayoutEdge> reversedPreds = new ArrayList<>();
+            for (LayoutEdge e : node.preds) {
+                if (e.relativeTo == pos && reversedLinks.contains(e.link)) {
+                    if (hasReversedDown) {
+                        e.relativeTo = curX - offset;
+                    } else {
+                        e.relativeTo = node.width + offset;
+                    }
+
+                    reversedPreds.add(e);
+                }
+            }
+            node.height += offset;
+            ArrayList<Point> endPoints = new ArrayList<>();
+
+            node.width += offset;
+            if (hasReversedDown) {
+                curX -= offset;
+                endPoints.add(new Point(curX, node.height));
+            } else {
+                curX += offset;
+                endPoints.add(new Point(node.width, node.height));
+            }
+
+            node.outOffsets.put(pos - minX, curX);
+            curX += offset;
+            node.bottomYOffset += offset;
+
+            endPoints.add(new Point(pos, node.height));
+            endPoints.add(new Point(pos, oldNodeHeight));
+            for (LayoutEdge e : reversedPreds) {
+                reversedLinkEndPoints.put(e.link, endPoints);
+            }
+        }
+
+        if (minX < 0) {
+            for (LayoutEdge e : node.preds) {
+                e.relativeTo -= minX;
+            }
+
+            for (LayoutEdge e : node.succs) {
+                e.relativeFrom -= minX;
+            }
+
+            node.xOffset = -minX;
+            node.width += -minX;
```

Comment:
Somewhat confusing. Rather use:
```suggestion
            node.width -= minX;
```